google-cloud-storage/src/test/java/com/google/cloud/storage/StorageImplMockitoTest.java

```diff
@@ -1637,4 +1637,459 @@ public void testWriterFailure() {
+        new StorageRpc.RewriteResponse(rpcRequest, null, 42L, false, "token", 21L);
+    doReturn(rpcResponse)
+        .doThrow(UNEXPECTED_CALL_EXCEPTION)
+        .when(storageRpcMock)
+        .openRewrite(rpcRequest);
+    initializeService();
+    CopyWriter writer = storage.copy(request);
+    assertEquals(42L, writer.getBlobSize());
+    assertEquals(21L, writer.getTotalBytesCopied());
+    assertTrue(!writer.isDone());
+  }
+
+  @Test
+  public void testCopyMultipleRequests() {
+    Storage.CopyRequest request =
+        Storage.CopyRequest.of(BLOB_INFO1.getBlobId(), BLOB_INFO2.getBlobId());
+    StorageRpc.RewriteRequest rpcRequest =
+        new StorageRpc.RewriteRequest(
+            request.getSource().toPb(),
+            EMPTY_RPC_OPTIONS,
+            false,
+            BLOB_INFO2.toPb(),
+            EMPTY_RPC_OPTIONS,
+            null);
+    StorageRpc.RewriteResponse rpcResponse1 =
+        new StorageRpc.RewriteResponse(rpcRequest, null, 42L, false, "token", 21L);
+    StorageRpc.RewriteResponse rpcResponse2 =
+        new StorageRpc.RewriteResponse(rpcRequest, BLOB_INFO1.toPb(), 42L, true, "token", 42L);
+    doReturn(rpcResponse1)
+        .doThrow(UNEXPECTED_CALL_EXCEPTION)
+        .when(storageRpcMock)
+        .openRewrite(rpcRequest);
+
+    doReturn(rpcResponse2)
+        .doThrow(UNEXPECTED_CALL_EXCEPTION)
+        .when(storageRpcMock)
+        .continueRewrite(rpcResponse1);
+    initializeService();
+    CopyWriter writer = storage.copy(request);
+    assertEquals(42L, writer.getBlobSize());
+    assertEquals(21L, writer.getTotalBytesCopied());
+    assertTrue(!writer.isDone());
+    assertEquals(expectedBlob1, writer.getResult());
+    assertTrue(writer.isDone());
+    assertEquals(42L, writer.getTotalBytesCopied());
+    assertEquals(42L, writer.getBlobSize());
+  }
+
+  @Test
+  public void testCopyFailure() {
+    Storage.CopyRequest request =
+        Storage.CopyRequest.of(BLOB_INFO1.getBlobId(), BLOB_INFO2.getBlobId());
+    StorageRpc.RewriteRequest rpcRequest =
+        new StorageRpc.RewriteRequest(
+            request.getSource().toPb(),
+            EMPTY_RPC_OPTIONS,
+            false,
+            BLOB_INFO2.toPb(),
+            EMPTY_RPC_OPTIONS,
+            null);
+    doThrow(STORAGE_FAILURE).when(storageRpcMock).openRewrite(rpcRequest);
+    initializeService();
+    try {
+      storage.copy(request);
+      fail();
+    } catch (StorageException e) {
+      assertSame(STORAGE_FAILURE, e.getCause());
+    }
+
+    StorageRpc.RewriteResponse rpcResponse =
+        new StorageRpc.RewriteResponse(rpcRequest, null, 42L, false, "token", 21L);
+    doReturn(rpcResponse)
+        .doThrow(UNEXPECTED_CALL_EXCEPTION)
+        .when(storageRpcMock)
+        .openRewrite(rpcRequest);
+    initializeService();
+    CopyWriter writer = storage.copy(request);
+    assertEquals(42L, writer.getBlobSize());
+    assertEquals(21L, writer.getTotalBytesCopied());
+    assertTrue(!writer.isDone());
```

Comment:
assertFalse

Reply:
@elharo assertTrue changed to assertFalse when needed, please take a look.

Reply:
But now it causes linkage-monitor failing:

```
* Connection #0 to host storage.googleapis.com left intact
+ java -jar linkage-monitor-latest-all-deps.jar com.google.cloud:libraries-bom
Error: Exception in thread "main" java.lang.IllegalArgumentException: The option 'send-analytics' contains an illegal character : '-'
	at org.apache.commons.cli.OptionValidator.validateOption(OptionValidator.java:71)
	at org.apache.commons.cli.Option.<init>(Option.java:146)
	at org.apache.commons.cli.Options.addOption(Options.java:150)
	at org.apache.commons.cli.Options.addOption(Options.java:131)
	at com.google.cloud.tools.dependencies.linkagemonitor.LinkageMonitor.parseCommandLine(LinkageMonitor.java:124)
	at com.google.cloud.tools.dependencies.linkagemonitor.LinkageMonitor.main(LinkageMonitor.java:96)
Error: Process completed with exit code 1.
```

Reply:
@elharo could you take a look, please